### ISSUE TITLE
downsampling results in 0 or 1 raster without float

### ISSUE THREAD

**Issue by pz-max** [PyPSA/atlite#238]
## Description

Pictures tell more than words

## Expected Behavior

![image](https://user-images.githubusercontent.com/61968949/170465228-16d128f1-f223-41da-9583-e691f25e29a3.png)

## Actual Behavior

![image](https://user-images.githubusercontent.com/61968949/170465324-d0bf4b97-f470-452a-8922-96d00ffee96b.png)

## Your Environment

* The `atlite` version used:
* How you installed `atlite` (`conda`, `pip` or `github`):
* Operating System:
* My environment:
    <details>
      <summary>(output of `conda list`)</summary>
      ```
        <!-- output of `conda list` -->
      ```
    </details>

![image](https://user-images.githubusercontent.com/61968949/170465659-4ba8009c-e3c7-471f-9803-7d6a1ad1c35c.png)

**Comment by pz-max** [comment]
Trying to debug this if others face the same issue

**Comment by pz-max** [comment]
I am trying to understand why the cutout raster has either a 0 or a 1 availability and nothing in between... ( as the expected image behaviour would suggest). I was looking now into the function `shape_availability_reprojected()` in `gis.py`. Here in particular [`rasterio.warp.reproject()`](https://rasterio.readthedocs.io/en/latest/api/rasterio.warp.html#rasterio.warp.reproject) is used to downsample (high-resolution raster -> low-resolution raster). 

Looking into the variables:
- **masked** represents the fine-grained shape e.g. in case of croatia an array/raster of the shape (4876, 5065)
- **dst_shape** represents the coarse-grained shape e.g. in case of croatia an array of the shape (31, 45)
- **resampling** represents the method of how we upsample the shape. *5* is method "average". There a many more methods available ([here](https://rasterio.readthedocs.io/en/latest/api/rasterio.enums.html#rasterio.enums.Resampling)). One can be here also more explicit like replacing **"5"** with **"rasterio.enums.Resampling.average"** (tested and works)
- the other cars stuff indicated the CRS of each raster & the transformation (relation between the raster resolutions) is required for upsampling...

So here I am with a bit more knowledge but haven't found the cause of the problem. Maybe it has something to do with the "**src_nodata**" stuff described [here](https://rasterio.readthedocs.io/en/latest/api/rasterio.warp.html#rasterio.warp.reproject). :)

https://github.com/PyPSA/atlite/blob/e3d0865afc4bcf6011f368ecfff9d67eefbfad84/atlite/gis.py#L473-L519

**Comment by FabianHofmann** [comment]
@pz-max thank you very much for testing this, this was extremely important. I found the error. Rasterio changed the order of the resampling methods (without a warning). I'll push a fix with a safe method selection.

**Comment by pz-max** [comment]
I can confirm it is still not working. :)

**Comment by pz-max** [comment]
Fixed the issue PR is coming @FabianHofmann

**Comment by pz-max** [comment]
## Reproduce the error
```python
git clone https://github.com/PyPSA/atlite.git
cd atlite
conda env create -f environment.yaml  # will install rasterio 1.2.10
pip install atlite  # install to test the examples
pytest
```
Leads to errors
![image](https://user-images.githubusercontent.com/61968949/170771491-a6df6adb-d1c7-4fdf-b339-185aa2d014ed.png)
Which I could reproduce in the examples:
![image](https://user-images.githubusercontent.com/61968949/170772388-3c453ff9-9d0d-4a11-9230-bd37cd14bd25.png)

**Comment by pz-max** [comment]
Performing then  `pip install 'rasterio<1.2.10'` fixes the issue (downgrades to 1.2.9)
![image](https://user-images.githubusercontent.com/61968949/170772960-27865890-c064-43de-9ff5-cd92800a4130.png)

**Comment by pz-max** [comment]
constraining rasterio in the `environment.yaml` to
```python
- rasterio>=1.0,<1.2.10  # until 1.3.0 is released
``` 
solves the issue we experienced (even though the anaconda installation quite a bit longer). The solved rasterio version is 1.2.8 (yes not 1.2.9)
![image](https://user-images.githubusercontent.com/61968949/170776709-4004bf68-5ac1-4419-a9bf-a4334a395ed2.png)

I checked the pytest to confirm everything runs well
![image](https://user-images.githubusercontent.com/61968949/170776895-622b1882-e053-4748-8d41-b4be27802848.png)

One note, the **rasterio** team has a [pre-release in the beta (v.1.3.0)](https://github.com/rasterio/rasterio/releases) with several improvements that also impact the downsampling/upsampling. This beta version also solves our issues (I tested it). Once fully released we should change our package constraints.
